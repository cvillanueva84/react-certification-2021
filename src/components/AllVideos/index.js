export { default } from './AllVideos.component';