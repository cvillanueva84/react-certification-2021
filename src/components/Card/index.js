export {default} from './Card.component.jsx'; 