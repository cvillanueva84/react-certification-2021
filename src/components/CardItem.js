import React from 'react';
import { useContext } from 'react';
import { GlobalContext } from './Context';
import {CardItemUi}  from './ui/CardItemUi'

export const CardItem = ({ title, description, url, videoId }) => {

  const { myStateReducer, dispatch } = useContext(GlobalContext);

  const handleClick = () => {
      dispatch({
        type: 'actionAddVideoData',
        payload: {
          ...myStateReducer,
          videoData: {
            title: title,
            description: description,
            url: url,
            videoId: videoId
          }
        }
      });
  }

  return (
    <CardItemUi 
      title={title} 
      description={description} 
      url={url} 
      handleClick={handleClick}
    />
  );

};

