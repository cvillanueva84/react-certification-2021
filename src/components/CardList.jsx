import React from 'react';
import Grid from '@material-ui/core/Grid';
import VideoCard from './VideoCard';
import videos from '../youtube-videos-mock';

export default function CardList() {
  return (
    <Grid
      container
      justifyContent="center"
      alignItems="center"
      style={{ padding: '30px' }}
    >
      {videos.items.map((video) => (
        <VideoCard videoInfo={video.snippet} key={video.etag} />
      ))}
    </Grid>
  );
}
