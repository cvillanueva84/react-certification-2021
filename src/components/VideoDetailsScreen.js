import React, { useContext } from 'react'
import { useFetchVideos } from '../hooks/useFetchVideos';
import { GlobalContext } from './Context';
import { useHistory } from 'react-router';
import { VideoDetailsScreenUi } from './ui/VideoDetailsScreenUi';

export const VideoDetailsScreen = () => {

    const { myStateReducer, dispatch } = useContext(GlobalContext);
    const { items: videos } = useFetchVideos(myStateReducer.search);
    const history = useHistory();

    const addFavorite = () => {
        dispatch({
            type: 'actionAddFavorites',
            payload: {
                ...myStateReducer,
                favorites: [myStateReducer.videoData, ...myStateReducer.favorites]
            }
        });
        history.replace('/favorites');
    }

    return (
        <VideoDetailsScreenUi 
            myStateReducer={myStateReducer} 
            videos={videos} 
            addFavorite={addFavorite}
        />
    )
}
