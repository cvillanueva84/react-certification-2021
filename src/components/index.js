export { default as Navbar } from './Navbar/Navbar';
export { default as SearchBar } from './SearchBar/SearchBar';
export { default as ThemeSwitch } from './ThemeSwitch/ThemeSwitch';
export { default as Results } from './Results/Results';
