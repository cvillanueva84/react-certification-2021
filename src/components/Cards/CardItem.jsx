import React from "react";
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
// import IconButton from '@material-ui/core/IconButton';
// import FavoriteIcon from '@material-ui/icons/Favorite';
import Typography from '@material-ui/core/Typography';
import CardActionArea from '@material-ui/core/CardActionArea';

import useStyles from './styles'


function CardItem({ img, title, description }) {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardActionArea>
        <CardMedia
          className={classes.media}
          image={img}
          title={title}
        />
        <CardContent>
          <Typography gutterBottom variant="p" component="h2" align="left">
            {title}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p" align="left">
            {description}
          </Typography>
        </CardContent>
        <CardActions>
          {/* <IconButton aria-label="add to favorites">
            <FavoriteIcon />
          </IconButton> */}
        </CardActions>
      </CardActionArea>
      
    </Card>
  );
}

export default CardItem;
