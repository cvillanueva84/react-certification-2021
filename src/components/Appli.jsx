import React from 'react';
import CardList from './CardList';
import NavBar from './NavBar';

export default function Appli() {
  return (
    <div>
      <NavBar />
      <CardList />
    </div>
  );
}
