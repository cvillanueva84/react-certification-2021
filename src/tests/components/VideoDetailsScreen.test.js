import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router';
import { VideoDetailsScreen } from '../../components/VideoDetailsScreen';
import { GlobalContext } from '../../components/Context';
import { CardItem } from '../../components/CardItem';

describe('Test VideoDetailsScreen', () => {
    const contextValue = {
        dispatch: jest.fn(),
        myStateReducer: {
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }
    }
    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const wrapper = mount(
        <GlobalContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <VideoDetailsScreen>
                        <CardItem/>
                    </VideoDetailsScreen>
                </Router>
            </MemoryRouter>
        </GlobalContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    });  

    it('should render success', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.exists());
    });

    it('should have elements', () => {
        expect(wrapper.getElements(".video-details").length).toBe(1);
        expect(wrapper.getElements("frame").length).toBe(1);
        expect(wrapper.getElements(".video-details-text").length).toBe(1);
        expect(wrapper.getElements("h2").length).toBe(1);
        expect(wrapper.getElements("p").length).toBe(1);
        expect(wrapper.getElements(".contenedor-sugeridos").length).toBe(1);
    });


    it('should dispatch addFavorite', () => {
        wrapper.find('#addFavorite').simulate('click');
        expect(contextValue.dispatch).toHaveBeenCalled();
    });



});