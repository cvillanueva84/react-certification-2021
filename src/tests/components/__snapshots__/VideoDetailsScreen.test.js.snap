// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test VideoDetailsScreen should render success 1`] = `ReactWrapper {}`;
